### ISSUE TITLE
exception in progress::aborted on wx backend

### ISSUE THREAD

**Issue by m32** [VPROFi/sqlplugin#1]
when this plugin is loaded on linux with wx backend exception is generated in this method, perhaps because on this backend console input is not stdin:
HANDLE std_in = stdin;

more portable is (i hope):
    WORD EscCode = VK_ESCAPE;
    if (WINPORT(CheckForKeyPress)(NULL, &EscCode, 1, CFKP_KEEP_OTHER_EVENTS))
		return true;
	return false;

**Comment by VPROFi** [comment]
We can't use **CheckForKeyPress**, because  [it wait for press key,](https://github.com/VPROFi/far2l/blob/89cff0e879ba09efdfa8141b52256d715c4b019e/WinPort/src/APIConsole.cpp#L214) **PeekConsoleInput** _**don't wait**_ for key.

You can send backtrace of exception, because `std_in = stdin` not used on linux in PeekConsoleInput. You can use 0 in this position.

**Comment by m32** [comment]
it is problematic to choose stdin as a handle to the event source, a backend other than tty generates an exception (wx).
it worked for me HANDLE std_in = NULL;
because far2l selects the default event source for the backend

CheckForKeyPress waits, but by default it waits for 0 ms, i.e. it does not wait. Progress based on this method does not hang, but only checks whether one of the mentioned keys has been pressed. At least on linux with wx as backend.

**Comment by VPROFi** [comment]
>> **You can send backtrace of exception**



> it worked for me HANDLE std_in = NULL;




This is good fix, if you send me **"backtrace of exception"**, we would see that the exception is associated with (for example) the impossibility of importing "stdin". If so, then the fix is needed not only in this plugin. If possible, please send a backtrace.


about waiting - some bug in CheckForKeyPress
```
STEP 1 CurrentPriority() return 0
unsigned int ConsoleInput::CurrentPriority() const
{
	if (_requestor_priorities.empty())
		return 0;

	return *_requestor_priorities.rbegin();
}
STEP 2 WaitForNonEmptyWithTimeout(0) return true
class ConsoleInput : public IConsoleInput
{
public:
	virtual bool WaitForNonEmptyWithTimeout(unsigned int timeout_msec, unsigned int requestor_priority = 0);
...
};

bool ConsoleInput::WaitForNonEmptyWithTimeout(unsigned int timeout_msec, unsigned int requestor_priority)
{
	std::unique_lock<std::mutex> lock(_mutex);
	for (;;) {
		if (!_pending.empty() && requestor_priority >= CurrentPriority())
			return true;
...
}
STEP 3 while condition true in CheckForKeyPress (may be bug)
WINPORT_DECL(CheckForKeyPress,DWORD,(HANDLE hConsoleInput, const WORD *KeyCodes, DWORD KeyCodesCount, DWORD Flags))
{
...
while (g_winport_con_in->WaitForNonEmptyWithTimeout(0)) {
			INPUT_RECORD rec;
			if (!g_winport_con_in->Dequeue(&rec, 1)) {
				break;
			}

we waiting in Dequeue in InspectCallbacks(data, i, true);
2 problems:
1) we can wait
2) we take away the press of a key that does not belong to us
```

**Comment by VPROFi** [comment]
in old versions of far2l before 2.6 hConsoleInput **is ignored** (which is correct for WinAPI emulation - GetStdHandle(STD_INPUT_HANDLE) returns non-zero)
```c
	WINPORT_DECL(PeekConsoleInput,BOOL,(HANDLE hConsoleInput, PINPUT_RECORD lpBuffer, DWORD nLength, LPDWORD lpNumberOfEventsRead))
	{
		*lpNumberOfEventsRead = g_winport_con_in->Peek(lpBuffer, nLength);
		return TRUE;
	}
```
in the new version 2.6+ **NOT IGNORED**, which is INCORRECT for WinAPI emulation and if you pass non-zero - then we will fly out to ASSERT
```c
static IConsoleInput *ChooseConIn(HANDLE hConsole)
{
	if (!hConsole) {
		return g_winport_con_in;
	}
	ForkedConsole *fc = (ForkedConsole *)hConsole;
	ASSERT(fc->magic == FORKED_CONSOLE_MAGIC);
	return fc->con_in;
}

	WINPORT_DECL(PeekConsoleInput,BOOL,(HANDLE hConsoleInput, PINPUT_RECORD lpBuffer, DWORD nLength, LPDWORD lpNumberOfEventsRead))
	{
		*lpNumberOfEventsRead = ChooseConIn(hConsoleInput)->Peek(lpBuffer, nLength);
		return TRUE;
	}

```

**Comment by VPROFi** [comment]
[fix сompatibility with version 2.6.3, bypass incorrect emulation PeekConsoleInput](https://github.com/VPROFi/sqlplugin/commit/c855546abcd3e08044ebb4af5dead436e740f0f0)
